sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercionSuite.scala

```diff
@@ -366,7 +366,29 @@ class TypeCoercionSuite extends AnalysisTest {
     // No up-casting for fixed-precision decimal (this is handled by arithmetic rules)
     widenTest(DecimalType(2, 1), DecimalType(3, 2), None)
     widenTest(DecimalType(2, 1), DoubleType, None)
-    widenTest(DecimalType(2, 1), IntegerType, None)
+    widenTest(DecimalType(2, 1), IntegerType, Some(DecimalType(11, 1)))
```

Comment:
We should have one and only one positive and negative test case for each integral type(byte, short, int, long), and another positive and negative test case for negative scale with int type.

Reply:
@cloud-fan OK.